### PR TITLE
Fix pointer to jlong  conversion in 32 bit OS

### PR DESCRIPTION
**Pull request by sike-evolve**

### CODE REVIEW

**Comment by siying** [comment]
@adamretter can you help take a look?

**Comment by adamretter** [comment]
@siying Sorry, I should have mentioned here, @sike-evolve is working with me, and I did already review it internally here - I will add a public review here too.

**Review by adamretter — APPROVED** [review]
LGTM

**Comment by adamretter** [comment]
@sike-evolve Can you rebase this please to resolve the conflict?

**Comment by sike-evolve** [comment]
Will do

File: java/rocksjni/cplusplus_to_java_convert.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#ifndef _CPLUSPLUS_TO_JAVA_CONVERT_H
```
Comment (siying) [inline java/rocksjni/cplusplus_to_java_convert.h:1]: This file misses copyright header.

File: java/rocksjni/cplusplus_to_java_convert.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#define _CPLUSPLUS_TO_JAVA_CONVERT_H
```
Comment (siying) [inline java/rocksjni/cplusplus_to_java_convert.h:2]: Is there a reason we don't use
```
#pragma once
```
as other header files?

**Review by siying — CHANGES_REQUESTED** [review]

File: java/rocksjni/cplusplus_to_java_convert.h (hunk 1 above)
Comment (adamretter) [reply java/rocksjni/cplusplus_to_java_convert.h]: Thanks @siying I have now added that

File: java/rocksjni/cplusplus_to_java_convert.h (hunk 2 above)
Comment (adamretter) [reply java/rocksjni/cplusplus_to_java_convert.h]: Thanks @siying I have now switched it to use that

**Comment by adamretter** [comment]
@siying I think this one is now ready to go.

**Comment by adamretter** [comment]
@jay-zhuang @mrambacher Can we get this one merged please?

**Pull request closed without merging**